### ISSUE TITLE
Problem with subtitles

### ISSUE THREAD

**Issue by Ph4te** [NineWorlds/serenity-android#304]
I'm using OpenMediaVault (basically a Debian) with Plex Media Server and my video files are stored in a share. Serenity shows all subtitles which were automatically downloaded by Plex, but all SRT-subtitles which were stored manually next to the video files are recognized by the Plex client but not the Serenity client. Waiting for 20 seconds doesn't help. Does anyone have any solutions for this problem?

**Comment by kingargyle** [comment]
Can you get provide some sample XML for the affected movies/tv shows.

You can get more info on how to obtain the xml from here:

https://github.com/NineWorlds/serenity-android/wiki/Serenity-for-Android#bugreports

Also what version of the Plex server are you using?

**Comment by Ph4te** [comment]
Thanks for your answer.

Version of Plex-Server:
      version="0.9.12.18.1520-6833552"

For example the problem exists with this one. My Plex client recognises the subtitles, but Serenity doesn't:

     <Directory ratingKey="3472" key="/library/metadata/3472/children" studio="CBS" type="show"   title="Extant" contentRating="TV-14" summary="Extant is an American science fiction television drama series created by Mickey Fisher and executive produced by Steven Spielberg. The story revolves around astronaut Molly Woods (Halle Berry) who returns home to her family inexplicably pregnant after 13 months in outer space on a solo mission." index="1" rating="7.5" lastViewedAt="1447100357" year="2014" thumb="/library/metadata/3472/thumb/1436570955" art="/library/metadata/3472/art/1436570955" banner="/library/metadata/3472/banner/1436570955" theme="/library/metadata/3472/theme/1436570955" duration="3600000" originallyAvailableAt="2014-07-09" leafCount="13" viewedLeafCount="0" childCount="1" addedAt="1436568132" updatedAt="1436570955"><Genre tag="Drama"/><Genre tag="Science-Fiction"/><Role tag="Halle Berry"/><Role tag="Goran Visnjic"/><Role tag="Pierce Gagnon"/></Directory>

**Comment by kingargyle** [comment]
Okay, to get the subtitle information we'll have to go deeper into the network.  Can you navigate to the following path:

/library/metadata/3472/children

and get me the XML from there.

**Comment by Ph4te** [comment]
Here you go:

/library/metadata/3472/children

     <MediaContainer size="1" allowSync="1" art="/library/metadata/3472/art/1436570955" banner="/library/metadata/3472/banner/1436570955" identifier="com.plexapp.plugins.library" key="3472" librarySectionID="2" librarySectionTitle="TV Shows" librarySectionUUID="8f17487f-5820-47b6-bb7a-17f28021a76d" mediaTagPrefix="/system/bundle/media/flags/" mediaTagVersion="1446075544" nocache="1" parentIndex="1" parentTitle="Extant" parentYear="2014" summary="Extant is an American science fiction television drama series created by Mickey Fisher and executive produced by Steven Spielberg. The story revolves around astronaut Molly Woods (Halle Berry) who returns home to her family inexplicably pregnant after 13 months in outer space on a solo mission." theme="/library/metadata/3472/theme/1436570955" thumb="/library/metadata/3472/thumb/1436570955" title1="TV Shows" title2="Extant" viewGroup="season" viewMode="65593"><Directory ratingKey="3473" key="/library/metadata/3473/children" parentRatingKey="3472" type="season" title="Season 1" parentKey="/library/metadata/3472" summary="" index="1" lastViewedAt="1447100357" thumb="/library/metadata/3473/thumb/1436570955" leafCount="13" viewedLeafCount="0" addedAt="1436568132" updatedAt="1436570955">
</Directory></MediaContainer>

**Comment by kingargyle** [comment]
Hmmm... that xml doesn't contain what I was lookng for, I may have given you the wrong path to extract.  Let me take a look at the code to see what section the subtitle look is looking for.

**Comment by kingargyle** [comment]
Alright looks like we need to go down at last another level or two to get to the actual show meta data.

Can you send the XML from:

/library/metadata/3473/children

That should give us a larger amount of XML, and then from there we'll need to get one more list that actually contains the full list of metadata that contains the subtitle info.

**Comment by kingargyle** [comment]
Sorry for the super late response,  neither of these contained the necessary XML.  I'm going to start work on some updates so I'll try and retest this and get some more information out on the particular XML that is needed.  I also need to test against the latest Plex servers.

**Comment by Ph4te** [comment]
No problem, I'll just wait. I would have to wait anyway.

**Comment by Ph4te** [comment]
I mostly use the original Plex-App now, so this problem didn't really bother me anymore. I just did use Serenity and I got the solution by chance.

If I use Serenity, it's always with the folder view, in which one file is shown above the other. It didn't really occur to me to click on a single season poster on the bottom. When I did, the single episodes were displayed one at a time. And in this folder view I can choose subtitles.
